File: Subarray sum equals K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> sumMap;
        int sum=0, i=0, count=0;
        while(i<nums.size()){
            sum+=nums[i];
            if(sum==k)
                count++;
            if(sumMap.find(sum-k)!=sumMap.end())
                count+= sumMap[sum-k];
            sumMap[sum]+=1;
            i++;
        }
        return count;
    }
};

File: Subarray Sum Equals K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        sums = defaultdict(int)
        s = 0
        subarray = 0

        for num in nums:
            s += num
            if s == k:
                subarray += 1
            subarray += sums[s-k]
            sums[s] += 1

        return subarray 

